test/test_interpreter.cpp

```diff
@@ -651,9 +650,9 @@ TEST_SUITE("xsystem_clone")
     {
         xcpp::xsystem system;
 
-        xcpp::xpreamble* clone = system.clone();
+        std::unique_ptr<xcpp::xpreamble> clone = system.clone();
 
-        REQUIRE(dynamic_cast<xcpp::xsystem*>(clone) != nullptr);
+        REQUIRE(dynamic_cast<xcpp::xsystem*>(clone.get()) != nullptr);
```

Comment:
Does `xsystem` have multiple inheritance? If not we do not need `dynamic_cast`s here.

Reply:
Addressed.